packages/admin/cms-admin/src/generator/future/generateForm.ts

```diff
@@ -41,7 +42,9 @@ export function generateForm(
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const formFields = config.fields.reduce<FormFieldConfig<any>[]>((acc, field) => {
-        if (isFormLayoutConfig(field)) {
+        if (field.type === "fieldSet") {
+            acc.push(...field.fields.filter(isFormFieldConfig));
```

Comment:
Why do you need this if? Didn't you plan to achieve this using form layouts?

Reply:
because fieldSet can also contain optionalNestedFields-Type typescript complained. I'll check again if I find a solution to solve this without checking for fieldSet-type.

Reply:
I found a way without typescript error.

Reply:
maybe this should also be refactored. Moving getting formFields into return-param of generateFields.

Currently `const rootBlocks`-object (containing block-type, input2state and defaultValues-function), extracting `readOnlyFields` before update-mutation (could maybe be implemented with `omitFromFormValues`) and getForwardedGqlArgs-feature (could already depend on returned `formFragmentFields`) depends on this list...